packages/xstate-react/src/useMachine.ts

```diff
@@ -59,16 +62,30 @@ export interface UseMachineOptions<TContext, TEvent extends EventObject> {
 export function useMachine<
   TContext,
   TEvent extends EventObject,
-  TTypestate extends Typestate<TContext> = { value: any; context: TContext }
+  TTypestate extends Typestate<TContext> = { value: any; context: TContext },
+  TMeta extends TypegenMeta = DefaultTypegenMeta
 >(
-  getMachine: MaybeLazy<StateMachine<TContext, any, TEvent, TTypestate>>,
-  options: Partial<InterpreterOptions> &
-    Partial<UseMachineOptions<TContext, TEvent>> &
-    Partial<MachineOptions<TContext, TEvent>> = {}
+  ...[getMachine, options = {}]: TMeta extends OptionsAreRequired
+    ? [
+        getMachine: MaybeLazy<
+          StateMachine<TContext, any, TEvent, TTypestate, any, TMeta>
+        >,
+        options: Partial<InterpreterOptions> &
+          Partial<UseMachineOptions<TContext, TEvent>> &
+          MaybeRequiredTypegenMachineOptions<TContext, TEvent, TMeta>
+      ]
+    : [
+        getMachine: MaybeLazy<
+          StateMachine<TContext, any, TEvent, TTypestate, any, TMeta>
+        >,
+        options?: Partial<InterpreterOptions> &
+          Partial<UseMachineOptions<TContext, TEvent>> &
+          MaybeRequiredTypegenMachineOptions<TContext, TEvent, TMeta>
+      ]
```

Comment:
Nit: same here (simplification) Worried about how this will read on hover for end-users too

Reply:
Because the members of the tuple are labelled, it actually reads really nicely (and normally)